src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -1112,6 +1112,12 @@ static int client_receive_message(
         assert(client->event);
 
         buflen = next_datagram_size_fd(fd);
+        if (buflen == -ENETDOWN) {
+                /* the link is down. Don't return an error or the I/O event
```

Comment:
you mean when link is up we we don;t process anything to DHCP. Please add a test test to make sure this indeed works .

Reply:
This does not look like affect networkd . since we stop the client when link goes down. But NM is having issues we can merge this. But I think the more time the client will run there will be a TMO and even if link comes back DHCP client won't be able to send immediately discoverer if the lease expired in this period some seconds.